datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxHorizontalScalingService.java

```diff
@@ -99,46 +121,83 @@ public String triggerScalingFlow(SdxCluster sdxCluster, StackScaleV4Request scal
+            throw new BadRequestException(String.format("The Data lake cannot be scaled in the %s environment status.",
+                    environmentResponse.getEnvironmentStatus().name()));
         }
         StackV4Response stackResponse = stackV4Endpoint.getByCrn(1L, sdxCluster.getStackCrn(), Set.of());
-        String nodeRequestValidity = isNodeRequestValid(stackResponse, request);
-        if (StringUtils.isNotEmpty(nodeRequestValidity)) {
-            throw new BadRequestException(String.format("The requested nodeCount does not reach the minimum condition of horizontal scaling. Issues: %s",
-                    nodeRequestValidity));
+        if (YARN == stackResponse.getCloudPlatform()) {
+            throw new BadRequestException("The Data lake horizontal scaling is not supported on YCloud");
+        }
+        String nodeRequestValid = isNodeRequestValid(stackResponse, request);
+        if (StringUtils.isNotEmpty(nodeRequestValid)) {
+            throw new BadRequestException(nodeRequestValid);
         }
     }
 
     private String isNodeRequestValid(StackV4Response stack, DatalakeHorizontalScaleRequest request) {
         if (request.getDesiredCount() < 0) {
             LOGGER.warn("Negative nodeCount was requested.");
-            return "Negative nodeCount is not accepted. If you want to downsacle your cluster please use lower number than the acutal node count";
+            return "Negative nodeCount is not accepted. If you want to downsacle your Data lake please use lower number than the acutal node count";
         }
         String target = request.getGroup();
-        boolean hostGroupNotExists = stack.getInstanceGroups().stream()
-                .map(InstanceGroupV4Response::getName)
-                .noneMatch(target::equalsIgnoreCase);
-        if (hostGroupNotExists) {
+        String gatewayInstanceName = DatalakeInstanceGroupScalingDetails.GATEWAY.getName().toLowerCase(Locale.ROOT);
+        Map<String, Set<InstanceMetaDataV4Response>> instanceMap = getInstanceMap(stack, target, gatewayInstanceName);
+        if (!instanceMap.containsKey(target)) {
             LOGGER.warn("The requested hostgroup name not found!");
-            return String.format("The requested hostgroup name not found! The requested host group name is %s", target);
+            return String.format("The requested hostgroup name not found! The requested hostgroup name is %s", target);
+        }
+        if (!instanceMap.containsKey(gatewayInstanceName)) {
+            LOGGER.error("Gateway node is not found in the Stack Instance Map. StackId: {}", stack.getId());
+            List<String> instanceGroupList = stack.getInstanceGroups().stream().map(InstanceGroupV4Base::getName).toList();
+            return String.format("The Gateway instance not found, found instances: %s", instanceGroupList);
+        }
+        List<InstanceMetaDataV4Response> gatewayMetaData = getGatewayMetaData(gatewayInstanceName, instanceMap);
```

Comment:
The intent is to validate the gateway node where CM is running and not all the nodes in the gateway host group.